### PR TITLE
Removed controls animation during transitions

### PR DESCRIPTION
**Pull request by RomanTysiachnik**
## Removed controls animation during transitions
I've added a new flag - controlsAnimationEnabled that now will depend on:
- `uiProps` value;
- `State` struct with next values:
     - `hasAppeared` - if the view has appeared ;
     - `isTransitioning` - if transition in proccess.
     - `controlsAnimationPossible` -  affects the adding of animations to controls layers. It will be true, if the view has appeared and if view isn't transitioning.

**Broken:**
broken

**Fixed:**
fixed
[JIRA Issue](https://jira.ops.aol.com/browse/OMSDK-700)

### CODE REVIEW

File: PlayerControls/sources/DefaultControlsViewController.swift [hunk 1]
```diff
@@ -107,6 +107,15 @@ public final class DefaultControlsViewController: ContentControlsViewController
         isApperared = true
     }
     
+    var isTransition = false
```
Comment (AndriiMoskvin) [inline PlayerControls/sources/DefaultControlsViewController.swift]: `isTransitioning` ?

File: PlayerControls/sources/DefaultControlsViewController.swift [hunk 2]
```diff
@@ -132,7 +141,11 @@ public final class DefaultControlsViewController: ContentControlsViewController
             afterFadeAnimationActions.append(block)
         }
 
-        if uiProps.animationsEnabled {
+        var controlsAnimationEnabled: Bool {
+            return uiProps.animationsEnabled && isApperared && !isTransition
```
Comment (AndriiMoskvin) [inline PlayerControls/sources/DefaultControlsViewController.swift]: `hasAppeared`?

Comment (RomanTysiachnik) [reply PlayerControls/sources/DefaultControlsViewController.swift]: Right.

File: PlayerControls/sources/DefaultControlsViewController.swift [hunk 3]
```diff
@@ -107,6 +107,15 @@ public final class DefaultControlsViewController: ContentControlsViewController
         isApperared = true
     }
     
+    var isTransition = false
+    public override func viewWillTransition(to size: CGSize, with coordinator: UIViewControllerTransitionCoordinator) {
+        isTransition = true
+        coordinator.animateAlongsideTransition(in: nil, animation: nil) { _ in
```
Comment (BogdanBilonog) [inline PlayerControls/sources/DefaultControlsViewController.swift]: Why before `super` call?

Comment (RomanTysiachnik) [reply PlayerControls/sources/DefaultControlsViewController.swift]: Missed it. Fixed.

**Comment by RomanTysiachnik** [comment]
I'm looking forward to approval.

File: PlayerControls/sources/DefaultControlsViewController.swift [hunk 4]
```diff
@@ -132,7 +141,11 @@ public final class DefaultControlsViewController: ContentControlsViewController
             afterFadeAnimationActions.append(block)
         }
 
-        if uiProps.animationsEnabled {
+        var controlsAnimationEnabled: Bool {
+            return uiProps.animationsEnabled && hasAppeared && !isTransitioning
```
Comment (BogdanBilonog) [inline PlayerControls/sources/DefaultControlsViewController.swift]: Maybe better to move `hasAppeared` and `isTransitioning` inside some structure and on `didSet` call `setNeedsLayout()`

Comment (RomanTysiachnik) [reply PlayerControls/sources/DefaultControlsViewController.swift]: You mean a struct with 2 fields and each one rewrites controlsAnimationEnabled and triggers `setNeedsLayout` on didSet?

Comment (AndriiMoskvin) [reply PlayerControls/sources/DefaultControlsViewController.swift]: Not sure that I understand
> each one rewrites controlsAnimationEnabled

Comment (RomanTysiachnik) [reply PlayerControls/sources/DefaultControlsViewController.swift]: @AndriiMoskvin I guess I didn't understand either.

File: PlayerControls/sources/DefaultControlsViewController.swift [hunk 5]
```diff
@@ -101,10 +101,27 @@ public final class DefaultControlsViewController: ContentControlsViewController
         }
     }
     
-    var isApperared = false
+    private var controllerStates = ControllerStates()
```
Comment (BogdanBilonog) [inline PlayerControls/sources/DefaultControlsViewController.swift]: `States`? Is it collection?

Comment (RomanTysiachnik) [reply PlayerControls/sources/DefaultControlsViewController.swift]: @BogdanBilonog how should I call it then?

Comment (BogdanBilonog) [reply PlayerControls/sources/DefaultControlsViewController.swift]: @RomanTysiachnik  `ControllerState`?

File: PlayerControls/sources/DefaultControlsViewController.swift [hunk 6]
```diff
@@ -101,10 +101,27 @@ public final class DefaultControlsViewController: ContentControlsViewController
         }
     }
     
-    var isApperared = false
+    private var controllerState = ControllerState()
+    struct ControllerState {
+        var hasAppeared: Bool = false
+        var isTransitioning: Bool = false
```
Comment (AndriiMoskvin) [inline PlayerControls/sources/DefaultControlsViewController.swift]: `Bool` not needed here

File: PlayerControls/sources/DefaultControlsViewController.swift [hunk 7]
```diff
@@ -101,10 +101,27 @@ public final class DefaultControlsViewController: ContentControlsViewController
         }
     }
     
-    var isApperared = false
+    private var controllerState = ControllerState()
```
Comment (AndriiMoskvin) [inline PlayerControls/sources/DefaultControlsViewController.swift]: Maybe just `state`?

File: PlayerControls/sources/DefaultControlsViewController.swift [hunk 8]
```diff
@@ -101,10 +101,27 @@ public final class DefaultControlsViewController: ContentControlsViewController
...
+    struct ControllerState {
+        var hasAppeared: Bool = false
+        var isTransitioning: Bool = false
+        
+        var controlsAnimationPossible: Bool {
```
Comment (AndriiMoskvin) [inline PlayerControls/sources/DefaultControlsViewController.swift]: Why `possible`? `allowed` sounds better for me.

Comment (RomanTysiachnik) [reply PlayerControls/sources/DefaultControlsViewController.swift]: I thought that `possible`means that it is not possible to animate controls unless the view has appeared and that current layout is not forced by the transition.

File: PlayerControls/sources/DefaultControlsViewController.swift (hunk 7 above)
Comment (RomanTysiachnik) [reply PlayerControls/sources/DefaultControlsViewController.swift]: Maybe.

File: PlayerControls/sources/DefaultControlsViewController.swift (hunk 6 above)
Comment (RomanTysiachnik) [reply PlayerControls/sources/DefaultControlsViewController.swift]: Got it.

File: PlayerControls/sources/DefaultControlsViewController.swift (hunk 8 above)
Comment (AndriiMoskvin) [reply PlayerControls/sources/DefaultControlsViewController.swift]: Ok, leave at is.

File: PlayerControls/sources/DefaultControlsViewController.swift (hunk 5 above)
Comment (RomanTysiachnik) [reply PlayerControls/sources/DefaultControlsViewController.swift]: @BogdanBilonog fixed

**Pull request merged**